src/SapNwRfc/SapServer.cs

```diff
@@ -149,31 +150,55 @@ private static RfcResultCode HandleGenericFunction(RfcInterop interop, Action<IS
             }
         }
 
-        private static RfcResultCode HandleGenericMetadata(RfcInterop interop, SapConnectionParameters parameters, string functionName, out IntPtr funcDescHandle)
+        private static RfcResultCode HandleGenericMetadata(string functionName, RfcAttributes attributes, out IntPtr funcDescHandle, Func<string, SapAttributes, ISapFunctionMetadata> getFunctionMetadata)
```

Comment:
If the changes in this PR are solely to remove some duplication here, I would not merge it. 

But I understand that it is to avoid opening connections each time? I wonder, do we have to get the metadata from the connection each time the server requests it, or could we get the metadata once and return that everytime the server asks for it. What would be the downside of this? (I never used server callbacks before, so please bear with me 😅)

Reply:
This PR exposes the `getFunctionMetadata` to the user, so they have more control over retrieving the metadata. So they can:
- Handle exceptions on their own.
- Reuse `SapConnection`s.
- Keep control over the `SapConnectionParameters`.
- Use different connections depending on `functionName` and `SapAttributes`. Which might be needed if there are multiple `SapServer` instances connected to different SAP systems. Since `InstallGenericServerFunctionHandler` is static after all.